R/construct_design.R

```diff
@@ -62,60 +56,110 @@
 #'
 #' summary(diagnosis)
 #' }
-#'
-declare_design <- function(...) {
+#' 
+#' @importFrom rlang enexpr expr_deparse
+#' 
+#' @export
+`+.dd` <- function(lhs, rhs){
+  
+  # two cases
+  # 1. lhs is a step
+  # 2. lhs is a design
+  
+  browser()
+  
+  if (!inherits(rhs, "dd")) {
+    stop("The right hand side does not appear to be a dd object. Can you wrap the step with `tidy_step()`?", call. = FALSE)
+  }
+  
+  if (inherits(lhs, "design")) {
+    
+    # browser()
+    auto_steps <- sapply(lhs, function(x) { lbl <- attr(x, "label"); !is.null(lbl) && grepl("Autogenerated by ", lbl) })
```

Comment:
add a subclass or other metadata attribute to the autogenerated steps, don't rely on labels.

Reply:
thanks, makes sense, will do